### DISCUSSION TITLE
How do I use Terser's "keep_quoted" option in my next.js app without breaking the code?

### DISCUSSION THREAD

**Discussion by JayJohn1869** [vercel/next.js#13660] [Help]
Hello!

I have the following config that sets the `keep_quoted` propery in `mangle` to true. ([See mangle options](https://github.com/terser/terser#mangle-properties-options))
```javascript
webpack(config) {
  for (const plugin of config.optimization.minimizer) {
    if (plugin['constructor']['name'] === 'TerserPlugin') {
      plugin.options.terserOptions.mangle.properties = {
        keep_quoted: 'strict'
      }
    }
  }
  
  return config
}
```
This causes the code to break. Thankfully terser accepts a `reserved` option which is an array of identifiers that will not have their properties mangled.

Is there a list of all the properties that need to go into this array so that the code functions?
Note that I'm looking for identifiers used by next.js and react, since my objects are already using the bracket syntax to access property names if needed!

Thanks!

**Comment by Timer** [comment]
Can you please provide a code example that breaks?

**Comment by JayJohn1869** [comment]
Even a starter file, like this one breaks:
```javascript
export default function App() {
  return (
    <h1>Hello</h1>
  )
}
```

My guess is that terser mangles react/next.js internal properties as well, so that code like `t.constructor` is mangled into `t.a`, which ends up breaking internals

**Reply by Timer** [reply]
This example is not sufficient to cause a break. This works locally and in our CI.

Please provide a full demo.

**Reply by JayJohn1869** [reply]
You can take a look at this sandbox:
https://codesandbox.io/s/happy-stallman-u4t45

Notice the only thing I changed was the `next.config.js` file

**Reply by Timer** [reply]
Please remove that `next.config.js`. You should not be configuring Terser, and are causing this broken behavior yourself.

**Reply by JayJohn1869** [reply]
Understood, is there a way to mangle class properties without configuring terser? Perhaps a babel plugin?

**Reply by Timer** [reply]
Mangling class properties is nearly 100% guaranteed to break code. Usage cannot be statically inferred.

**Reply by t1gor** [reply]
@Timer @JayJohn1869 Hello, guys. Did you manage to find the solution for the issue? Having same trouble but with the lib in my case - can't really change it.
